### ISSUE TITLE
Private property that starts with `_` and is refed by getter function should be ignored

### ISSUE THREAD

**Issue by Goooler** [pinterest/ktlint#2332]
## Expected Behavior

This should be ignored?

```kotlin
class Foo {
  private val _foo = "foo"

  fun getFoo(): String = _foo
}
```

If this `_foo` is refed by `foo` property, it works, it should also work with the referring of getter function?

## Observed Behavior

Using CLI:
```sh
ktlint Foo.kt        
/Users/goooler/IdeaProjects/licensee/Foo.kt:2:15: Property name should start with a lowercase letter and use camel case (standard:property-naming)

Summary error count (descending) by rule:
  standard:property-naming: 1
```

EditorConfig file:
```editorconfig
root = true

[*]
charset = utf-8
indent_size = 2
indent_style = space
insert_final_newline = true
trim_trailing_whitespace = true

[*.{kt,kts}]
ij_kotlin_imports_layout = *
ij_kotlin_allow_trailing_comma = true
ij_kotlin_allow_trailing_comma_on_call_site = true
ktlint_code_style = intellij_idea
```

## Steps to Reproduce

Run `ktlint Foo.kt`.

## Your Environment

* Version of ktlint used: 1.0.1
* Relevant parts of the `.editorconfig` settings
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task):
* Version of Gradle used (if applicable):
* Operating System and version:

**Comment by paul-dingemans** [comment]
See [property names](https://kotlinlang.org/docs/coding-conventions.html#property-names) and [names for backing properties](https://kotlinlang.org/docs/coding-conventions.html#names-for-backing-properties) in Kotlin Coding conventions. Your `fun getFoo()` is not a property.

**Comment by Goooler** [comment]
But if we support getter property like:

```kotlin
class Foo {
  private val _foo = "foo"

  val foo: String get() = _foo
}
```
We can also support getter function like:
```kotlin
class Foo {
  private val _foo = "foo"

  fun getFoo(): String = _foo
}
```

They have the same signature.

**Comment by paul-dingemans** [comment]
The current implementation is just very rudimentary and stricly follows the convention. So it requires a `private` property with name `_someName` and another `public` property with the same name except that it does not start with `_` (so that is `someName`). It is not checked that the value of the `public` variable refers to the backing property itself.

So something similar can be done for your case. If a `public` function exists with name starting with `get` followed by initial case version of the property name without underscore, and not having any arguments then the backing property should not be reported.

Note: I just see in the code that it is not actually verified that the `public` property is really `public`. This is to be fixed in #2345
